ppr-ui/src/components/common/PidNumber.vue

```diff
@@ -0,0 +1,152 @@
+<template>
+  <v-row id="pid-number" class="pid-text-input" no-gutters>
+    <v-col cols="12">
+      <p class="font-weight-bold">PID Number</p>
+    </v-col>
+    <v-col cols="12" sm="3">
+      <v-text-field
+        id="pid-one-input"
+        ref="pidOneRef"
+        maxlength="3"
+        filled persistent-hint autofocus
+        hint="Parcel identifier must contain 9 digits"
+        :readonly="enablePidLoader"
+        :error-messages="invalidPidMsg"
+        v-model="pidOne"
+      />
+    </v-col>
+
+    <v-divider class="horizontal-divider pb-1" />
+
+    <v-col cols="12" sm="3">
+      <v-text-field
+        id="pid-two-input"
+        ref="pidTwoRef"
+        filled
+        maxlength="3"
+        :readonly="enablePidLoader"
+        v-model="pidTwo"
+      />
+    </v-col>
+
+    <v-divider class="horizontal-divider pb-1" />
+
+    <v-col cols="12" sm="3">
+      <v-text-field
+        id="pid-three-input"
+        ref="pidThreeRef"
+        filled
+        maxlength="3"
+        :readonly="enablePidLoader"
+        v-model="pidThree"
+      />
+    </v-col>
+
+    <v-col cols="12" sm="1">
+      <v-progress-circular
+        v-if="enablePidLoader"
+        indeterminate
+        color="primary"
+      ></v-progress-circular>
+    </v-col>
+  </v-row>
+</template>
+
+<script lang="ts">
+/* eslint-disable no-unused-vars */
+import vue from 'vue'
+import { computed, defineComponent, reactive, toRefs, watch } from '@vue/composition-api'
+import { useInputRules } from '@/composables'
+/* eslint-enable no-unused-vars */
+
+export default defineComponent({
+  name: 'PidNumber',
+  components: {},
+  props: {},
+  setup (props, context) {
+    // Composable(s)
+    const {
+      isNumber
+    } = useInputRules()
+
+    const localState = reactive({
+      pidOne: '',
+      pidTwo: '',
+      pidThree: '',
+      enablePidLoader: false,
+      pidNumber: computed(() => {
+        return `${localState.pidOne}-${localState.pidTwo}-${localState.pidThree}`
+      }),
+      isValidPid: computed(() => {
+        return (
+          (localState.pidOne ? /^\d+$/g.test(localState.pidOne) : true) &&
+          (localState.pidTwo ? /^\d+$/g.test(localState.pidTwo) : true) &&
+          (localState.pidThree ? /^\d+$/g.test(localState.pidThree) : true)
+        )
+      }),
+      invalidPidMsg: computed(() => {
+        return localState.isValidPid ? [] : ['Enter a valid PID']
+      })
+    })
+
+    const emitPid = (): void => { context.emit('setPid', localState.pidNumber) }
+
+    watch(() => localState.pidOne, () => {
+      // @ts-ignore - function exists
+      if (localState.pidOne.length === 3) vue.nextTick(() => { context.refs.pidTwoRef.focus() })
+    })
+    watch(() => localState.pidTwo, () => {
+      // @ts-ignore - function exists
+      if (localState.pidTwo.length === 3) vue.nextTick(() => { context.refs.pidThreeRef.focus() })
```

Comment:
Nice! What about going backwards when deleting pid numbers (e.g. from pid 3 to 2 to 1)?